### PR TITLE
Adding DI.External to runtime with its history

### PR DESCRIPTION
**Pull request by maryamariyan**

### CODE REVIEW

**Review by tarekgh — APPROVED** [review]

**Comment by tarekgh** [comment]
Thanks @maryamariyan for brining in these tests

File: src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkCurrent)</TargetFrameworks>
+    <EnableDefaultItems>true</EnableDefaultItems>
+    <SignAssembly>false</SignAssembly>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <Compile Include="..\..\..\Common\tests\Extensions\TestingUtils\Microsoft.AspNetCore.Testing\src\ExceptionAssertions.cs">
```
Comment (safern) [inline src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:10]: Use `CommonTestPath` MSBuild property.

File: src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj [hunk 2]
```diff
@@ -0,0 +1,39 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFrameworks>$(NetCoreAppCurrent);$(NetFrameworkCurrent)</TargetFrameworks>
+    <EnableDefaultItems>true</EnableDefaultItems>
+    <SignAssembly>false</SignAssembly>
```
Comment (safern) [inline src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:6]: Why can't we sign this assembly? We use a Test key.

Comment (tarekgh) [reply src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:6]: These tests are using 3ed parties packages which including non-signed assemblies.

File: src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj [hunk 3]
```diff
@@ -24,16 +22,12 @@
     <PackageReference Include="Newtonsoft.Json" Version="$(NewtonsoftJsonVersion)" />
     <PackageReference Include="Autofac.Extensions.DependencyInjection" Version="4.3.1.0"/>
```
Comment (safern) [inline src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj:23]: This version is weird, is it really `4.3.1.0` or just `4.3.1`?

File: src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj (hunk 2 above)
Comment (maryamariyan) [reply src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.External.Tests/Microsoft.Extensions.DependencyInjection.ExternalContainers.Tests.csproj]: Added NoWarn so we could sign this as a test assembly

**Comment by maryamariyan** [comment]
moved files slightly so that DI tests now contain 3 folders:

- DI.External.Tests (with its own csproj)
- DI.Specification.Tests (used in the other two)
- DI.Tests (with its own csproj)

**Comment by maryamariyan** [comment]
Failures on net472:
```
System.IO.FileLoadException : Could not load file or assembly 'Lamar, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. A strongly-named assembly is required. (Exception from HRESULT: 0x80131044)
   at Microsoft.Extensions.DependencyInjection.Specification.LamarDependencyInjectionSpecificationTests.CreateServiceProviderImpl(IServiceCollection serviceCollection)
   at Microsoft.Extensions.DependencyInjection.Specification.DependencyInjectionSpecificationTests.OuterServiceCanHaveOtherServicesInjected() in /_/src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Specification.Tests/DependencyInjectionSpecificationTests.cs:line 230

System.IO.FileLoadException : Could not load file or assembly 'DryIoc, Version=3.0.2.0, Culture=neutral, PublicKeyToken=null' or one of its dependencies. A strongly-named assembly is required. (Exception from HRESULT: 0x80131044)
   at Microsoft.Extensions.DependencyInjection.Specification.DryIocDependencyInjectionSpecificationTests.CreateServiceProvider(IServiceCollection serviceCollection)
   at Microsoft.Extensions.DependencyInjection.Specification.DependencyInjectionSpecificationTests.ServicesRegisteredWithImplementationTypeCanBeResolved() in /_/src/libraries/Microsoft.Extensions.DependencyInjection/tests/DI.Specification.Tests/DependencyInjectionSpecificationTests.cs:line 24
```

**Review by JunTaoLuo — APPROVED** [review]

**Comment by maryamariyan** [comment]
The CI already went green. I just force pushed my changes now so that my commits-only were squashed together, keeping the commit history intact, doing a merge commit to submit the PR.

**Pull request merged**